rules.bzl

```diff
@@ -127,3 +127,7 @@ def android_ndk_repository(name, **kwargs):
         **kwargs
     )
     native.register_toolchains("@%s//:all" % name)
+    native.bind(
```

Comment:
I think in general we are trying to get rid of bind()s, since they don't work with bzlmod. Is there another way to do this?

Reply:
Right now the default value for [`--android_crosstool_top` is `//external:android/crosstool`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/rules/android/AndroidConfiguration.java;drc=19f5e933d3fc91848b2b786cb11a6decaa96cf6e;l=213). So we either need to add the binding or change the flag default to be `@androidndk//:toolchain` (or something similar).

Reply:
If we make this repo support incompatible toolchain resolution we can probably ignore this? but otherwise users of this repo have to pass the flag to use this anyways? https://github.com/bazelbuild/rules_android_ndk/?tab=readme-ov-file#getting-started

Reply:
No, some change is required regardless of `--incompatible_enable_android_toolchain_resolution`, because the core problem is in toolchain resolution when a non-Android configured target tries to find the current SDK.

Reply:
Deleted previous comment: I am working on too many issues at once in this space.

I still see failures related to this when enabling toolchain resolution, because some code paths still require _something_ to be present. This change does remove the need to explicitly pass `--android_crosstool_top` (which I consider to be a benefit to users).

Reply:
Discussed this with @ted-xie today: Ideally we'd want to remove all uses of `bind`, but today there are several, and rules_android_ndk doesn't support bzlmod anyway, yet.

Let's get this in and get to feature parity with native, then deal with moving forward.